src/Command/CheckAllSchemasCompatibilityCommand.php

```diff
@@ -0,0 +1,154 @@
+            ->setHelp('Checks for compatibility for all schemas in folder')
+            ->addArgument('schemaDirectory', InputArgument::REQUIRED, 'Path to avro schema directory');
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     * @return integer
+     * @throws RequestException
+     */
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        /** @var string $directory */
+        $directory = $input->getArgument('schemaDirectory');
+        $avroFiles = $this->getAvroFiles($directory);
+
+        $io = new SymfonyStyle($input, $output);
+
+        $failed = [];
+
+        if (false === $this->checkSchemas($avroFiles, $failed)) {
+            $io->error('Following schemas are not compatible:');
+            $io->listing($failed);
+
+            return 1;
+        }
+
+        return 0;
+    }
+
+    /**
+     * @param string $directory
+     * @return array
+     */
+    protected function getAvroFiles(string $directory): array
+    {
+        $iterator = new RecursiveIteratorIterator(
+            new RecursiveDirectoryIterator(
+                $directory,
+                FilesystemIterator::SKIP_DOTS
+            )
+        );
+
+        $files = [];
+
+        /** @var SplFileInfo $file */
+        foreach ($iterator as $file) {
+            if (Avro::FILE_EXTENSION !== $file->getExtension()) {
+                continue;
+            }
+
+            $files[$file->getBasename('.' . Avro::FILE_EXTENSION)] = $file->getRealPath();
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param string $schemaName
+     * @param string $localSchema
+     * @param string $latestVersion
+     * @return boolean
+     */
+    protected function isLocalSchemaCompatible(
+        string $schemaName,
+        string $localSchema,
+        string $latestVersion
+    ): bool {
+        return $this->schemaRegistryApi->checkSchemaCompatibilityForVersion(
+            $localSchema,
+            $schemaName,
+            $latestVersion
+        );
+    }
+
+    /**
+     * @param string $schemaName
+     * @param string $localSchema
+     * @return boolean
+     */
+    protected function isAlreadyRegistered(
+        string $schemaName,
+        string $localSchema
+    ): bool {
+        $version = null;
+
+        try {
+            $version = $this->schemaRegistryApi->getVersionForSchema(
+                $schemaName,
+                $localSchema
+            );
+        } catch (Throwable $e) {
+        }
+
+        return null !== $version;
+    }
+
+    /**
+     * @param array $avroFiles
+     * @param array $failed
+     * @return boolean
+     */
+    private function checkSchemas(array $avroFiles, array &$failed = []): bool
+    {
+        $failed = [];
+
+        foreach ($avroFiles as $schemaName => $avroFile) {
+            /** @var string $fileContents */
+            $fileContents = file_get_contents($avroFile);
```

Comment:
```suggestion
            $localSchema = file_get_contents($avroFile);
```
I think this should be enough and we don't need the decode and encode steps below

Reply:
That was strange for me too, but I used it from refactored previous code and I guessed there is a strong reason for it.